build_tools/azure/install.sh

```diff
@@ -146,9 +140,10 @@ scikit_learn_install() {
         # TODO Always add --check-build-dependencies when all CI builds have
         # pip >= 22.1.1. At the time of writing, two CI builds (debian32_atlas and
         # ubuntu_atlas) have an older pip
-        if pip install --help | grep check-build-dependencies; then
-            ADDITIONAL_PIP_OPTIONS="$ADDITIONAL_PIP_OPTIONS --check-build-dependencies"
-        fi
+        # TODO temporary hack while waiting for version checks inside meson.build
+        # if [[ "$DISTRIB" == "ubuntu" || "$DISTRIB" == "debian-32" ]]; then
+        #     ADDITIONAL_PIP_OPTIONS="$ADDITIONAL_PIP_OPTIONS --check-build-dependencies"
+        # fi
```

Comment:
I don't understand the TODO comment: it's a temporary hack but actually it's commented out. Do you mean that we we would like to pass `--check-build-dependencies` but it's not possible because of a limitation of `meson.build`?

Reply:
Good point I forgot this, let me fix this.

Basically what happens following the Debian 32 bit Docker image update:
- `pip` is recent enough to have `--check-build-dependencies`
- system numpy is too old (1.24) to satisfy the `pyproject.toml` build dependencies (numpy>=1.25) so using `pip install --check-build-dependencies` will result in an error

Previously (before the Debian 32 bit Docker image update), `pip` was too old to have `--check-build-dependencies` flag ...

Reply:
I ended up removing this since there is no need to use `--check-build-dependencies` in the CI. This was mostly intended to have user friendly error when your build dependencies are not recent enough but it has limitation and #28721 will remove `--check-build-dependencies` everywhere soonish.